Source/problems/Castro_bc_fill_nd.cpp

```diff
@@ -109,17 +110,21 @@ void ca_statefill(Box const& bx, FArrayBox& data,
         (bcr[URHO].hi(0) == EXT_DIR &&           // xr, yr, zr corner
          (bcr[URHO].hi(1) == EXT_DIR || bcr[URHO].hi(2) == EXT_DIR)) ||
         (bcr[URHO].hi(1) == EXT_DIR && bcr[URHO].hi(2) == EXT_DIR)) {
-      amrex::Error("Error: external boundaries meeting at a corner not supported");
+        amrex::Error("Error: external boundaries meeting at a corner not supported");
     }
 #endif
 
     if (numcomp == 1) {
 
+#ifndef CXX_MODEL_PARSER
 #pragma gpu box(bx)
         ext_denfill(AMREX_INT_ANYD(bx.loVect()), AMREX_INT_ANYD(bx.hiVect()),
                     BL_TO_FORTRAN_N_ANYD(data, dcomp),
                     AMREX_INT_ANYD(geom.Domain().loVect()), AMREX_INT_ANYD(geom.Domain().hiVect()),
                     AMREX_REAL_ANYD(geom.CellSize()), AMREX_REAL_ANYD(geom.ProbLo()), time, bc_f);
+#else
+        ext_denfill_c(bx, data.array(), dcomp, geom, bcr[dcomp], time);
```

Comment:
The indexing into `bcr` is not consistent with the Fortran here -- `dcomp` is the destination component to be filled in the Fab, but `bcr` is a vector of length `numcomp` which are the BC values corresponding to components `dcomp` to `dcomp + numcomp - 1`.  So here you want to be doing `bcr[0]` here for consistency.

Reply:
fixed